src/guiguts/tools/pptxt.py

```diff
@@ -0,0 +1,2591 @@
+                    # Delete dash(es) just found. Any left at final test is 'unecognised'.
+                    line = line.replace(ch_em, "")
+                # If any dashes left on line at this point treat them as unrecognised
+                if re.search(r"\p{Pd}", line):
+                    a_un.append((line_index + 1, book[line_index]))
+
+            line_index += 1
+
+        # Generate dialog entries about remaining entries.
+
+        first_header = True
+
+        # If many pairs of "--" detected report only the first five.
+
+        if len(a_h2) > 0:
+            if first_header:
+                first_header = False
+            else:
+                checker_dialog.add_entry("")
+            checker_dialog.add_entry("Pairs of '--' (keyboard '-') found")
+
+            count = 0
+            for record in a_h2:
+                line_number = record[0]
+                line = record[1]
+
+                if count < 5:
+                    report_all_occurrences_on_line(r"(?<!-)--(?!-)", line, line_number)
+                count += 1
+            if count > 5:
+                if len(a_h2) - 5 == 1:
+                    output_record = "  ...1 more line"
+                else:
+                    output_record = f"  ...{len(a_h2) - 5} more lines"
+                checker_dialog.add_entry(output_record)
+            if counth2 > 5:
+                checker_dialog.add_entry("")
+                checker_dialog.add_entry(
+                    "    [Book seems to use '--' as em-dash so not reporting these further]"
+                )
+
+        # Report other consecutive dashes
+
+        if len(a_hh) > 0:
+            if first_header:
+                first_header = False
+            else:
+                checker_dialog.add_entry("")
+            checker_dialog.add_entry(
+                "Adjacent dashes (at least 8 emdash or keyboard '-' expected as separator)"
+            )
+
+            for record in a_hh:
+                line_number = record[0]
+                line = record[1]
+                report_all_occurrences_on_line(r"(\p{Pd}\p{Pd}+)", line, line_number)
+
+        # Report hyphen-minus
+
+        if len(a_hm) > 0:
+            if first_header:
+                first_header = False
+            else:
+                checker_dialog.add_entry("")
+            checker_dialog.add_entry("Hyphen-minus")
+            for record in a_hm:
+                line_number = record[0]
+                line = record[1]
+                report_all_occurrences_on_line(ch_hm, line, line_number)
+
+        # Report hyphen
+
+        if len(a_hy) > 0:
+            if first_header:
+                first_header = False
+            else:
+                checker_dialog.add_entry("")
+            checker_dialog.add_entry("Hyphen")
+            for record in a_hy:
+                line_number = record[0]
+                line = record[1]
+                report_all_occurrences_on_line(ch_hy, line, line_number)
```

Comment:
For a line such as
`This contains--a double hyphen and a single-hyphen`
each hyphen in the double hyphen is reported (unexpected), whereas for
`This only contains--a double hyphen`
the hyphens in the double hyphen are not reported (as expected).
This seems to be independent of whether the earlier check said "Book seems to use '--' as em-dash so not reporting these further"

Reply:
Here are a couple of screenshots of the issue. There should be just one message about line 1335, rather than three, because the first two are part of a double-hyphen, as in this screenshot:
![image](https://github.com/DistributedProofreaders/guiguts-py/assets/8938581/e69d741c-bcdf-4668-ad0a-804bb30297c4)
If I delete just the single hyphen from before "P2" on that line and re-run pptxt, then I get no messages at all about L1335, as in this screenshot:
![image](https://github.com/DistributedProofreaders/guiguts-py/assets/8938581/1aeef477-b0ea-4a25-9515-6d44ea53242c)

So, it seems that if there is a single hyphen, then every hyphen on the line gets reported (including those in double hyphens), but if there is no single hyphen, then hyphens within double hyphens are not reported.

Here's a zip of the text file the screenshots came from:
[projectID60849c915cbd1.zip](https://github.com/DistributedProofreaders/guiguts-py/files/14551099/projectID60849c915cbd1.zip)

Reply:
This is fixed in the latest version of PPtxt.